File: rxandroid/src/main/java/rx/android/widget/RxSeekBar.java
package rx.android.widget;

import android.widget.SeekBar;
import rx.Observable;

public final class RxSeekBar {
  /**
   * Create an observable of progress value changes on {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   */
  public static Observable<Integer> changes(SeekBar view) {
    return Observable.create(new SeekBarChangeOnSubscribe(view));
  }

  /**
   * Create an observable of progress change events for {@code view}.
   * <p>
   * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
   * to free this reference.
   */
  public static Observable<SeekBarChangeEvent> changeEvents(SeekBar view) {
    return Observable.create(new SeekBarChangeEventOnSubscribe(view));
  }

  private RxSeekBar() {
    throw new AssertionError("No instances.");
  }
}

File: rxandroid-kotlin/src/main/kotlin/rx/android/widget/RxSeekBar.kt
package rx.android.widget

import android.widget.SeekBar
import rx.Observable

/**
 * Create an observable of progress value changes on {@code view}.
 * <p>
 * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
 * to free this reference.
 */
fun SeekBar.changes(): Observable<Int> = RxSeekBar.changes(this)

/**
 * Create an observable of progress change events for {@code view}.
 * <p>
 * <em>Warning:</em> The created observable keeps a strong reference to {@code view}. Unsubscribe
 * to free this reference.
 */
fun SeekBar.changeEvents(): Observable<SeekBarChangeEvent> = RxSeekBar.changeEvents(this)
